recipes/oommf/build.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+TCLTKVERSION=8.5
+CONDA_ROOT_PATH=$(conda info --root)
+
+# Compile OOMMF
+export OOMMF_TCL_CONFIG=${PREFIX}/lib/tclConfig.sh
+export OOMMF_TK_CONFIG=${PREFIX}/lib/tkConfig.sh
+cd oommf
+tclsh${TCLTKVERSION} oommf.tcl pimake distclean
```

Comment:
It seems we run into a problem here, and I think putting the TCLTKVERSION  back into this bash script. Here are the observations:

## the error
<pre>
(testoommf) escape:~ fangohr$ oommf
<25728> oommf.tcl 1.2.1.0 Oc_Config warning:
Tcl version mismatch:
	/Users/fangohr/anaconda3/envs/testoommf/lib/tclConfig.sh from 8.5.18
	Running Tcl 8.5.9
(testoommf) escape:~ fangohr$ <25732> /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/omfsh/darwin/omfsh 1.2.1.0 panic:
Tcl initialization error:

Can't find a usable init.tcl in the following directories: 
    /System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts /opt/anaconda1anaconda2anaconda3/lib/tcl8.5 /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/omfsh/lib/tcl8.5 /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/lib/tcl8.5 /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/omfsh/library /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/library /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/tcl8.5.18/library /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/tcl8.5.18/library

/System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts/init.tcl: version conflict for package "Tcl": have 8.5.18, need exactly 8.5.9
version conflict for package "Tcl": have 8.5.18, need exactly 8.5.9
    while executing
"package require -exact Tcl 8.5.9"
    (file "/System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts/init.tcl" line 20)
    invoked from within
"source /System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts/init.tcl"
    ("uplevel" body line 1)
    invoked from within
"uplevel #0 [list source $tclfile]"


This probably means that Tcl wasn't installed properly.


As a workaround, set the environment variable
TCL_LIBRARY to the name of a directory
containing an error-free init.tcl.
</pre>

## Can reproduce with the last line of start up script:
<pre>
(testoommf) escape:~ fangohr$ which oommf
/Users/fangohr/anaconda3/envs/testoommf/bin/oommf
(testoommf) escape:~ fangohr$ cat `which oommf`
#! /bin/bash
export OOMMF_TCL_CONFIG=/Users/fangohr/anaconda3/envs/testoommf/lib/tclConfig.sh
export OOMMF_TK_CONFIG=/Users/fangohr/anaconda3/envs/testoommf/lib/tkConfig.sh
tclsh /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/oommf.tcl "$@"
</pre>

and then

<pre>
(testoommf) escape:~ fangohr$ tclsh /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/oommf.tcl "$@"
(testoommf) escape:~ fangohr$ <25889> /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/omfsh/darwin/omfsh 1.2.1.0 panic:
Tcl initialization error:

Can't find a usable init.tcl in the following directories: 
    /System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts /opt/anaconda1anaconda2anaconda3/lib/tcl8.5 /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/omfsh/lib/tcl8.5 /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/lib/tcl8.5 /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/omfsh/library /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/library /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/app/tcl8.5.18/library /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/tcl8.5.18/library

/System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts/init.tcl: version conflict for package "Tcl": have 8.5.18, need exactly 8.5.9
version conflict for package "Tcl": have 8.5.18, need exactly 8.5.9
    while executing
"package require -exact Tcl 8.5.9"
    (file "/System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts/init.tcl" line 20)
    invoked from within
"source /System/Library/Frameworks/Tcl.framework/Versions/8.5/Resources/Scripts/init.tcl"
    ("uplevel" body line 1)
    invoked from within
"uplevel #0 [list source $tclfile]"


This probably means that Tcl wasn't installed properly.


As a workaround, set the environment variable
TCL_LIBRARY to the name of a directory
containing an error-free init.tcl.
</pre>

## Fix
But using `tclsh8.5` instead of `tclsh` fixes the issue
<pre>
(testoommf) escape:~ fangohr$ tclsh8.5 /Users/fangohr/anaconda3/envs/testoommf/opt/oommf/oommf.tcl "$@"
(testoommf) escape:~ fangohr$ 2017-05-29 19:16:12.234 omfsh[25895:2080119]
</pre>
(This out output is good news; oommf working correctly.)

## diagnostics
<pre>
(testoommf) escape:~ fangohr$ which tclsh8.5
/Users/fangohr/anaconda3/envs/testoommf/bin/tclsh8.5
(testoommf) escape:~ fangohr$ which tclsh
/usr/bin/tclsh
(testoommf) escape:~ fangohr$ 
</pre>
This seems to suggest that tcl8.5 does not provide a symbolic link to `tclsh`, and that on this installation, the system tclsh is used instead of the conda one (resulting in the error).

## more info
<pre>
(testoommf) escape:~ fangohr$ conda info
Current conda install:

               platform : osx-64
          conda version : 4.2.13
       conda is private : False
      conda-env version : 4.2.13
    conda-build version : 2.1.13
         python version : 3.6.0.final.0
       requests version : 2.12.4
       root environment : /Users/fangohr/anaconda3  (writable)
    default environment : /Users/fangohr/anaconda3/envs/testoommf
       envs directories : /Users/fangohr/anaconda3/envs
          package cache : /Users/fangohr/anaconda3/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /Users/fangohr/.condarc
           offline mode : False

(testoommf) escape:~ fangohr$ conda list 
# packages in environment at /Users/fangohr/anaconda3/envs/testoommf:
#
oommf                     1.2b0_20160930b1               0    conda-forge
openssl                   1.0.2l                        0  
pip                       9.0.1                    py36_1  
python                    3.6.1                         2  
readline                  6.2                           2  
setuptools                27.2.0                   py36_0  
sqlite                    3.13.0                        0  
tk                        8.5.18                        0  
wheel                     0.29.0                   py36_0  
xz                        5.2.2                         1  
zlib                      1.2.8                         3  
(testoommf) escape:~ fangohr$ 
</pre>

## Output from creating that environment `testoommf`
<pre>
escape:~ fangohr$ conda create -n testoommf python=3
Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /Users/fangohr/anaconda3/envs/testoommf:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-1.0.2l             |                0         3.0 MB

The following NEW packages will be INSTALLED:

    openssl:    1.0.2l-0     
    pip:        9.0.1-py36_1 
    python:     3.6.1-2      
    readline:   6.2-2        
    setuptools: 27.2.0-py36_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py36_0
    xz:         5.2.2-1      
    zlib:       1.2.8-3      

Proceed ([y]/n)? y

Fetching packages ...
openssl-1.0.2l 100% |####################################################################################################| Time: 0:00:01   2.70 MB/s
Extracting packages ...
[      COMPLETE      ]|#######################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ]|#######################################################################################################################| 100%
#
# To activate this environment, use:
# > source activate testoommf
#
# To deactivate this environment, use:
# > source deactivate testoommf
#

escape:~ fangohr$ source activate testoommf
(testoommf) escape:~ fangohr$ conda install -c conda-forge oommf
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment /Users/fangohr/anaconda3/envs/testoommf:

The following NEW packages will be INSTALLED:

    oommf: 1.2b0_20160930b1-0 conda-forge

Proceed ([y]/n)? y

Linking packages ...
[      COMPLETE      ]|#######################################################################################################################| 100%
(testoommf) escape:~ fangohr$ which oommf
/Users/fangohr/anaconda3/envs/testoommf/bin/oommf
</pre>

## Now what?
So, should we go back to 
`+tclsh${TCLTKVERSION} oommf.tcl pimake distclean`
instead of 
`+tclsh oommf.tcl pimake distclean`?

Or 'fix'(?) the conda `tk` package? [This seems risky to me.]

Or have I missed another origin of this issue?

Views, @isuruf, @takluyver ?